File: rust/src/models/online_payment_method_config.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::online_methods_config::OnlineMethodsConfig;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct OnlinePaymentMethodConfig {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub config: Option<OnlineMethodsConfig>,
}

impl OnlinePaymentMethodConfig {
    pub fn new() -> Self {
        Self { config: None }
    }
}

File: typescript/src/models/onlinePaymentMethodConfig.ts
// this file is @generated
import {
  type OnlineMethodsConfig,
  OnlineMethodsConfigSerializer,
} from "./onlineMethodsConfig";

export interface OnlinePaymentMethodConfig {
  config?: OnlineMethodsConfig | null;
}

export const OnlinePaymentMethodConfigSerializer = {
  _fromJsonObject(object: any): OnlinePaymentMethodConfig {
    return {
      config:
        object["config"] != null
          ? OnlineMethodsConfigSerializer._fromJsonObject(object["config"])
          : undefined,
    };
  },

  _toJsonObject(self: OnlinePaymentMethodConfig): any {
    return {
      config:
        self.config != null
          ? OnlineMethodsConfigSerializer._toJsonObject(self.config)
          : undefined,
    };
  },
};
